qiskit/pulse/device_specification.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+Specification of the device.
+"""
+import logging
+from typing import List
+
+from qiskit.pulse.qubit import Qubit
+from qiskit.pulse.channels import DriveChannel, ControlChannel, MeasureChannel
+from qiskit.pulse.channels import AcquireChannel, MemorySlot, RegisterSlot
+
+logger = logging.getLogger(__name__)
+
+
+class DeviceSpecification:
+    """Implement a device specification, which is usually contructed from backend info."""
+
+    @classmethod
+    def create_from(cls, backend):
+        """
+        Create device specification with values in backend configuration.
+        Args:
+            backend(Backend):
+        """
+        config = backend.configuration()
+
+        # system size
+        n_qubit = config.n_qubits
+
+        # frequency information
+        qubit_lo_freqs = config.defaults['qubit_freq_est']
+        qubit_lo_range = config.qubit_lo_range
+        meas_lo_freqs = config.defaults['meas_freq_est']
+        meas_lo_range = config.meas_lo_range
+
+        # generate channels with assuming their numberings are aligned with qubits
+        drives = [DriveChannel(i, qubit_lo_freqs[i]) for i in range(n_qubit)]   # TODO: lo_ranges
+        controls = [ControlChannel(i) for i in range(n_qubit)]
```

Comment:
There is no guarantee that there is a control channel for each qubit. The only way this may be determined is from the `n_uchannels`. Furthermore, this is no guarantee that a control channel will be associated with a specific qubit.  For example, this could be a global magnetic field. The only information that exists is the `u_channel_lo` and the device Hamiltonian which is likely not sufficient to tie to a qubit.

Reply:
You're right. This implementation only works for currently-available backends since it assumes n_uchannels==n_qubits. I'd like to assign several uchannels to a qubit, but unfortunately current Backend have no information about which qubit connected with a uchannel. Control channel will not be associated with a specific qubit but it will be associcated with a specific HW components like qubits or buses, am I correct? In the near future, we will add `Bus` class (and more HW components if necessary) to deal with other types of devices. To do those, we will need to get more  information about device specification from backend. This will also be a helpful information for users designing their pulses.